### PR TITLE
[RadioButton] UIKit implementation

### PR DESCRIPTION
**Pull request by michael-zimmermann**
# What was implemented
A UIKit component for a RadioButton and a RadioButtonGroup:
- RadioButtonUIGroupView.swift
- RadioButtonUIView.swift

## How did you implement the feature
The feature implements a RadioButtonToggle group according to the [figma designs](https://www.figma.com/file/0QchRdipAVuvVoDfTjLrgQ/Spark-Specs-and-Handoff?node-id=127-6137&t=ajwXYdYsR8cGlaCO-0).
The interface for both the RadioButtonUIGroup and the RadioButtionUIView are the same as the SwiftUI components.
The changed value is propagated to the consumer of the component by a binding.
The use-cases and view-model of the SwiftUI component are re-used. All published variables of the view-model are subscribed to by the view, to update the view on any changes. This is a simulation of SwiftUI.

## How can we verify the feature
The feature can be verified by running the demo app.

## Screenshot
Simulator Screen Shot - iPhone 14 Pro - 2023-05-12 at 12 24 35

### CODE REVIEW

File: core/Sources/Components/RadioButton/View/RadioButtonViewModel.swift [hunk 1]
```diff
@@ -94,6 +94,12 @@ final class RadioButtonViewModel<ID: Equatable & CustomStringConvertible>: Obser
             self.stateDidUpdate()
         }
     }
+
+    func colorsFor(selectedID: ID) -> RadioButtonColors {
+        return self.useCase
+            .execute(theme: self.theme, state: self.state, isSelected: selectedID == self.id)
```
Comment (janniklasfreundt) [inline core/Sources/Components/RadioButton/View/RadioButtonViewModel.swift:100]: Is it correct that the colors are re-created here every time and not stored in `self.colors`? Isn't `self.colors` out-of-date?

File: core/Sources/Components/RadioButton/View/UIKit/RadioButtonUIView.swift [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        }
+    }
+
+    private func subscribeTo<Value>(_ publisher: some Publisher<Value, Never>, action: @escaping (Value) -> Void ) {
+        publisher
+            .receive(on: RunLoop.main)
+            .sink { value in
+                action(value)
+            }
+            .store(in: &self.subscriptions)
```
Comment (janniklasfreundt) [inline core/Sources/Components/RadioButton/View/UIKit/RadioButtonUIView.swift:188]: 👍

File: core/Sources/Components/RadioButton/View/UIKit/RadioButtonUIView.swift [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    static let toggleViewSpacing: CGFloat = 4
+    static let toggleViewPadding: CGFloat = 6
+    static let textLabelTopSpacing: CGFloat = 3
+}
+
+/// A radio button view composed of a toggle item, a label and a possible sublabel.
+/// The color of the view is determined by the state. A possible sublabel is also part of the state.
+/// The value of the radio button is represented by the generic type ID.
+/// When the radio button is selected, it will change the binding value.
+public final class RadioButtonUIView<ID: Equatable & CustomStringConvertible>: UIView {
```
Comment (janniklasfreundt) [inline core/Sources/Components/RadioButton/View/UIKit/RadioButtonUIView.swift:25]: Aren't the RadioButtonUIViewTests missing? (For SwiftUI there are RadioButtonViewTests)

File: core/Sources/Components/RadioButton/View/RadioButtonViewModel.swift (hunk 1 above)
Comment (michael-zimmermann) [reply core/Sources/Components/RadioButton/View/RadioButtonViewModel.swift:100]: This was a horrible little hack, to update the color. I might try and get rid of it again, but my first efforts didn't work.

Comment (michael-zimmermann) [reply core/Sources/Components/RadioButton/View/RadioButtonViewModel.swift:100]: I managed to be able to get rid of it 👍

File: core/Sources/Components/RadioButton/View/UIKit/RadioButtonUIView.swift (hunk 3 above)
Comment (michael-zimmermann) [reply core/Sources/Components/RadioButton/View/UIKit/RadioButtonUIView.swift:25]: Good catch, will add

File: core/Sources/Components/RadioButton/View/RadioButtonViewModel.swift (hunk 1 above)
Comment (janniklasfreundt) [reply core/Sources/Components/RadioButton/View/RadioButtonViewModel.swift]: Nice 👍

File: core/Sources/Components/RadioButton/View/UIKit/RadioButtonUIGroupView.swift [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    private var selectedID: Binding<ID>
+    private let theme: Theme
+    private let items: [RadioButtonItem<ID>]
+    private let title: String?
+    private let bodyFontMetrics = UIFontMetrics(forTextStyle: .body)
+    private var itemConstraints = [NSLayoutConstraint]()
+
+    var spacing: CGFloat {
+        return self.bodyFontMetrics.scaledValue(
+            for: (self.theme.layout.spacing.xLarge - 16),
```
Comment (janniklasfreundt) [inline core/Sources/Components/RadioButton/View/UIKit/RadioButtonUIGroupView.swift:25]: Not sure about the "16"?

Comment (michael-zimmermann) [reply core/Sources/Components/RadioButton/View/UIKit/RadioButtonUIGroupView.swift:25]: Will correct and give it a proper name. I also hadn't migrated this to the new `ScaledUIMetric` which I'll do too.

**Review by janniklasfreundt — APPROVED** [review]
👍

**Review by janniklasfreundt — APPROVED** [review]

**Pull request merged**